// Proto file representing the EDB AdaptiveSettings class and derived classes

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";

// DEPRECATED: use SimulationSetupService instead
service AdaptiveSettingsService {
  // Gets the list of AdaptiveFrequencyData for the active adaptive type
  rpc GetAdaptiveFrequencyDataList(EDBObjMessage) returns (AdaptiveFrequencyDataListMessage) {}

  // Sets the list of AdaptiveFrequencyData for the active adaptive type
  rpc SetAdaptiveFrequencyDataList(SetAdaptiveFrequencyDataListMessage) returns (EDBObjMessage) {}
}

// Message representing AdaptiveFrequencyData
message AdaptiveFrequencyDataMessage {
  string adaptive_frequency = 1;
  string max_delta = 2;
  int64 max_passes = 3;
}

// Message for GetAdaptiveFrequencyDataList
message AdaptiveFrequencyDataListMessage {
  repeated AdaptiveFrequencyDataMessage adaptive_frequency_data = 1;
}

// Message for SetMeshOperations
message SetAdaptiveFrequencyDataListMessage {
  EDBObjMessage adaptive_settings = 1;
  AdaptiveFrequencyDataListMessage adaptive_frequency_data_list = 2;
}
